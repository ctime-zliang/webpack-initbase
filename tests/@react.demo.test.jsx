/*
	enzyme 渲染方式
		- shallow: 前渲染, 生成 vDom, 不渲染子组件, 只能对当前组件做断言
		- render: 静态渲染, 生成 html 字符串, 适用 Cheerio 分析字符串并返回 Cheerio 对象
		- mount: 完全渲染, 将组建渲染成真实完整的 DOM, 可测试 DOM API 和组建声明周期, 由 jsdom 提供模拟浏览器环境支持
 */

import React from 'react'
import { render as testingReactRender, screen } from '@testing-library/react'
import { render as enzymeRender } from 'enzyme'

const Main = () => {
	return (
		<>
			<div className="demo-jest-test">Demo Jest Test</div>
		</>
	)
}

describe(`React Demo Test 1`, () => {
	test('DOM Operation', () => {
		testingReactRender(<Main />)
		// screen.debug()
		const element = document.querySelector(`.demo-jest-test`)
		expect(element).toBeTruthy()
		expect(element.textContent).toBe(`Demo Jest Test`)
	})
})

describe(`React Demo Test 2`, () => {
	test('DOM Operation', () => {
		enzymeRender(<Main />)
		// screen.debug()
		const element = document.querySelector(`.demo-jest-test`)
		expect(element).toBeTruthy()
		expect(element.textContent).toBe(`Demo Jest Test`)
	})
})
