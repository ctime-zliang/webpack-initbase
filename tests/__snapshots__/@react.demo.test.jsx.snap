// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React Demo Test 2 DOM Operation 1`] = `
<Fragment>
  <div
    className="demo-jest-test"
  >
    Demo Jest Test
  </div>
</Fragment>
`;
